### PR TITLE
 CP-17161: Add 'using' statement around dialogs to ensure garbage collection

### PR DESCRIPTION
**Pull request by MihaelaStoica**

### CODE REVIEW

File: XenAdmin/Controls/Wlb/WlbReportView.cs [hunk 1]
```diff
@@ -499,12 +502,14 @@ public void ExecuteReport()
             catch (Exception ex)
             {
                 log.Debug(ex, ex);
-                new ThreeButtonDialog(
+                using (var dlg = new ThreeButtonDialog(
                    new ThreeButtonDialog.Details(
                        SystemIcons.Error,
                        ex.Message,
-                       Messages.XENCENTER)).ShowDialog(this);
-                return;
```
Comment (Frezzle) [inline XenAdmin/Controls/Wlb/WlbReportView.cs]: Was this return removed intentionally?

Comment (MihaelaStoica) [inline XenAdmin/Controls/Wlb/WlbReportView.cs]: Yes, because it was redundant.

File: XenAdmin/Wizards/NewSRWizard.cs [hunk 2]
```diff
@@ -701,13 +720,18 @@ private bool AskUserIfShouldContinue()
...
                             SystemIcons.Warning,
                             string.Format(Messages.ALREADY_ATTACHED_ELSEWHERE, _srToReattach.Name, Helpers.GetName(xenConnection), 
                             Text)),
                         ThreeButtonDialog.ButtonOK,
-                        ThreeButtonDialog.ButtonCancel).ShowDialog(this);
+                        ThreeButtonDialog.ButtonCancel))
+                        {
+                            dialogResult = dlg.ShowDialog(this);
+                        }
+                    return DialogResult.Yes == dialogResult;
```
Comment (Frezzle) [inline XenAdmin/Wizards/NewSRWizard.cs]: Should this use DialogResult.OK instead of DialogResult.Yes?

Comment (MihaelaStoica) [inline XenAdmin/Wizards/NewSRWizard.cs]: Yes, it should be DialogResult.OK, I'll change it.

**Comment by Frezzle** [comment]
Looks good. Merging. :weary:

**Pull request merged**